tls/s2n_resume.c

```diff
@@ -353,12 +362,6 @@ S2N_RESULT s2n_deserialize_resumption_state(struct s2n_connection *conn, struct
         }
     } else if (format == S2N_SERIALIZED_FORMAT_TLS13_V1) {
         RESULT_GUARD(s2n_tls13_deserialize_session_state(conn, psk_identity, from));
-        if (conn->mode == S2N_CLIENT) {
-            /* Free the client_ticket after setting a psk on the connection.
-             * This prevents s2n_connection_get_session from returning a TLS1.3
-             * ticket before a ticket has been received from the server. */
-            RESULT_GUARD_POSIX(s2n_free(&conn->client_ticket));
-        }
```

Comment:
We can safely move this up to s2n_client_deserialize_with_session_ticket because no other caller triggers this case:
* s2n_client_deserialize_with_session_id: https://github.com/aws/s2n-tls/blob/5cc827d228db8579b7ede3556332994778b19f82/tls/s2n_resume.c#L382 TLS1.3 does not support session ids, so we'd never get a TLS1.3 ticket that way. That's [enforced on get_session](https://github.com/aws/s2n-tls/blob/5cc827d228db8579b7ede3556332994778b19f82/tls/s2n_resume.c#L219).
* s2n_decrypt_session_ticket: https://github.com/aws/s2n-tls/blob/5cc827d228db8579b7ede3556332994778b19f82/tls/s2n_resume.c#L847 Only called by the server
* s2n_decrypt_session_cache: https://github.com/aws/s2n-tls/blob/5cc827d228db8579b7ede3556332994778b19f82/tls/s2n_resume.c#L924 Only called by the server